src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.CompilationTracker.cs

```diff
@@ -528,14 +529,16 @@ private async Task<Compilation> GetOrBuildDeclarationCompilationAsync(SolutionSe
                 ImmutableArray<SourceGeneratedDocumentState>? authoritativeGeneratedDocuments =
                     state.GeneratedDocumentsAreFinal ? state.GeneratedDocuments : null;
 
+                var nonAuthoritativeGeneratedDocuments = state.GeneratedDocuments;
```

Comment:
what's the intuition between hte names 'authoritative' and 'non-authoritative'?

Reply:
I interpreted authoritative (term existed already) as "values known to be correct". I added non-authoritative to mean "values we have that are not known to be correct". For the latter, the consumer is required to provide the necessary validation prior to using them.

Reply:
sgtm.

Reply:
@sharwell's interpretation of the terms matched my intent.